packages-backend/express/src/types.ts

```diff
@@ -25,6 +25,16 @@ export type TSessionMiddlewareOptions = {
 
   /* Options for the `jwksRsa.expressJwtSecret` */
   jwks?: Omit<ExpressJwtOptions, 'jwksUri'>;
+
+  // This middleware needs to resolve the request URL to know about the
+  // 'audience' it needs to use for the JWT.
+  // By default it assumes a nodejs IncomingMessage object as its first parameter,
+  // however this is not always the case (eg; AWS lambda functions).
+  // When providing this function you will be delegated to resolve the URL from
+  // the first parameter the middleware receives.
+  // Form example, in AWS lambda functions it will be an Event object
+  // (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a1260a1f3d40f239b53fd29effba594b0d1bee08/types/aws-lambda/trigger/api-gateway-proxy.d.ts#L12)
```

Comment:
Rewording to understand the issue :). Drop if not better.

```suggestion
  // In order to determine the 'audience' for the JWT the middleware needs to resolve the request URL.
  // As a default the middleware assumes takes the IncomingMessage of Node.js as its first parameter.
  // This default needs to be overwritten at times.  For instance in the case of AWS lambda functions.
  // Do to so this function, when provided, allows delegation of resolving the URL from the middleware's request.
  // In the case of an AWS lambda function an  Event object will be passed as defined here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a1260a1f3d40f239b53fd29effba594b0d1bee08/types/aws-lambda/trigger/api-gateway-proxy.d.ts#L12
```

Reply:
I ended up with this text when trying to merge both proposals:
```

  // In order to determine the 'audience' for JWT validation, the middleware needs
  // to resolve the request url.
  // By default, the middleware assumes it receives a Node.js IncomingMessage object
  // as its first parameter.
  // This default behaviour might not work for every scenario as, for example,
  // AWS lambda functions, where that first parameter is an Event object.
  // (Reference: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html)
  // To handle these situations, this option allows the delegation of the URL resolution
  // from the request-like object provided to the middleware as its first parameter.
```

What do you think about it?

Reply:
Best of both worlds. I like it. Thanks!

Reply:
Updated in this commit 8142081dec38c0af7bd1d5c81e0b84a73c1d30a4